src/bp-core/bp-core-update.php

```diff
@@ -1153,6 +1157,59 @@ function bb_update_to_1_5_6() {
 }
 
 /**
+ * x.x.x update routine.
+ *
+ * - Migrate group id to associate forum.
+ * - Update forum meta _bbp_group_ids, associated with group id.
+ *
+ * @since BuddyBoss x.x.x
+ *
+ * @return void
+ */
+function bb_update_to_1_5_9() {
+	// Return, when group or forum component deactive.
+	if ( ! bp_is_active( 'groups' ) || ! bp_is_active( 'forums' ) ) {
+		return;
+	}
+
+	// Get all forum associated groups.
+	$group_data = groups_get_groups(
+		array(
+			'per_page'   => -1,
+			'fields'     => 'ids',
+			'status'     => array( 'public', 'private', 'hidden' ),
+			'meta_query' => array(
+				'relation' => 'AND',
+				array(
+					'key'     => 'forum_id',
+					'value'   => 'a:0:{}',
+					'compare' => '!=',
+				),
+				array(
+					'key'     => 'forum_id',
+					'value'   => '',
+					'compare' => '!=',
+				),
+			),
+		)
+	);
+
+	$groups = empty( $group_data['groups'] ) ? array() : $group_data['groups'];
+
+	foreach ( $groups as $group_id ) {
+		$forum_ids = groups_get_groupmeta( $group_id, 'forum_id' );
+
+		if ( empty( $forum_ids ) ) {
+			continue;
+		}
+
+		$forum_id = array_shift( $forum_ids );
+
+		bbp_update_forum_group_ids( $forum_id, array( $group_id ) );
+	}
```

Comment:
@asaquzzamanbb Please update the migration it will break the current working if one forum assigns to multiple ones.

Reply:
@KartikSuthar 
Migration has been updated here https://github.com/buddyboss/buddyboss-platform/pull/2117/commits/f72c830994de3b58496fe6ba31c66dec21e80387